---
title: "R Notebook"
output: html_notebook
---

Scientific Question:
What mutations and alternate splicing differences in DNA sequence exist between different variants of the gene FRMD4A, and how does that impact the homology in the structures of the protein?

Background:
My gene of interest is the FERM Domain Containing 4A (FRMD4A) gene. Previous research has shown a correlation between miss regulation of the FRMD4A gene, in particularly up regulation, and oral squamous cell carcinoma. As an inspiring dentist I am interested in oral squamous cell carcinoma and ways to prevent it. By understanding which variants of the FRMD4A gene have mutated protein structures that alter the function of the FRMD4A gene we can target those genes for therapy and treat for oral squamous cell carcinoma from very early stages. 
DNA sequence of the four FRMD4A variants are extracted from the ncbi nucleotides database. The protein structures used for the homology modeling were extracted from SWISS protein and PDB.

Scientific Hypothesis: 
If the sequences of the four FRMD4A gene variants are altered through mutations and/or alternate splicing then they will exhibit differences in their protein structures.

Analysis:
First, Multiple sequence alignment (msa) was performed on the DNA sequence from 4 variants of the FRMD4A gene which were extracted from the ncbi nucleotides database as fasta files. 
This data was visualized using pretty print to show the aligned sequences in a publication worthy figure. 
Second, Homology modeling was performed on the protein structures of the four variants of the FRMD4A gene. In order to do so I extracted the amino acid sequences of each of the variants as FASTA proteins. I then used those amino acid sequences to model the structures of the variants using the SWISS protein database. The PDB database was used in order to find and extract a template structure to model my variants on. I used pymol to view my structures.
In order to visualize the homology between these protein structures I created a heat map in order to see how each protein model varied from the other and how much similarity they had. 

```{r}
library(Biostrings)
library(msa)
```
- Biostrings is used for uncoding and reading long strings of data. In this case biostrings is helpful in reading and maninpulating the nucelotide sequences. 

- msa is used for multiple sequence alignment and includes a function called "msa pretty print" that cleans up the alignment.

```{r}
system.file("tex", "texshade.sty", package="msa")

FRMD4A_unaligned <- readAAStringSet("Wildtype.fasta")

Kinase_MSA <- msa(FRMD4A_unaligned)

print(Kinase_MSA, show="complete")

```

```{r}
msaPrettyPrint(Kinase_MSA, output="tex", showNames="left", showLogo="top",
logoColors="rasmol", shadingMode="functional", shadingModeArg="structure",
showLegend=FALSE, askForOverwrite=FALSE)
```

Homology Modeling 

```{r}
library(bio3d)
```
bio3d is a package that allows us to analyze the structures of proteins.

```{r}
pdb <- read.pdb("1e5w.1.pdb")
print(pdb)
```

```{r}
head(pdb$atom)
```
```{r}
print(pdb$xyz)
```

```{r}
modes <- nma(pdb)
```

```{r}
plot(modes)
```

```{r}
cm <- dccm(modes)
plot (cm, sse=pdb)
```


```{r}
plot(modes, sse=pdb)
```

```{r}
modes.anm <- nma(pdb, ff= "anm")
```

```{r}
r <- rmsip(modes, modes.anm)
```

```{r}
plot(r)
```
```{r}

```


```{r}
plot.bio3d(modes$fluctuations, sse=pdb, sheet.col="orange", helix.col="purple", typ="l", lwd=3, ylab="Fluctuations from NMA (custom label)")
```

```{r}
ids <- c("3u8z", "5xq0", "4cgk", "1e5w")
```

```{r}
raw.files <- get.pdb(ids)
```

```{r}
files1 <- pdbsplit(raw.files, ids)
```

```{r}
library(msa)
pdbs <- pdbaln(files1, exefile='msa')
```

```{r}
summary(c(seqidentity(pdbs)))
```

```{r}
modes2 <- nma.pdbs(pdbs, fit = TRUE, outpath= NULL)
```

```{r}
plot(modes2, pdbs, type="h")
```
```{r}
print(modes2)
```

```{r}
modes2$rmsip
```

```{r}
heatmap(1-modes2$rmsip, symm = TRUE)
```


